### ISSUE TITLE
Documentation for installing from Source for 2.9.0 is incomplete/perhaps incorrect

### ISSUE THREAD

**Issue by weshinsley** [ckan/ckan#5625]
**CKAN version**
2.9.0

**Describe the bug**

Documentation issues - or possibly installation issues.

Following instructions to install 2.9.0 from source (https://docs.ckan.org/en/2.9/maintaining/installing/install-from-source.html) on a clean Ubuntu Server 20.04 VM does not lead to a working CKAN on port 5000. There are a few issues, some of which I have found work arounds for, but ultimately, no webpage is served on port 5000, and it is unclear whether to continue with deploying the server or not.

**Steps to reproduce**
Follow the steps on https://docs.ckan.org/en/2.9/maintaining/installing/install-from-source.html from a new installation of Ubuntu 20.04 Server. I am choosing the latest stable version in part 2(c), and running everything (postgres/solr) on the same single server. I am not installing anything optional (eg, datastore). See additional details below for details and suggestions on my experience so far.

**Expected behavior**
In point 9, "Open http://127.0.0.1:5000/ in a web browser, and you should see the CKAN front page." 

**Additional details**
In step (1) you say to install the solr-jetty package, but in step (5), the instructions tell you how to configure tomcat. So, do we go with Jetty or Tomcat?

First try: I stayed with solr-jetty.

In that case, I think step 5 should include jetty instructions, which I think are that you should
`sudo nano /etc/jetty9/start.ini` and replace `jetty.port=8080` with `jetty.port=8983`. I am not sure whether similar 8080->8983 changes are needed in `etc/jetty9/jetty-http.xml` or `etc/jetty9/jetty-spring.xml` where 8080 comes up, but I did so to be sure. Step 5.2 is the same - then `sudo service jetty9 restart`, before you try to browser to localhost:8983. It seems to work - you get the jetty placeholder there, and solr is correctly on localhost:8983/solr - although it looks like quite an old version.

When I finsh the steps and try `ckan -c /etc/ckan/default/ckan.ini run`,  the console reports:
```
2020-09-29 13:52:34,179 INFO  [ckan.cli] Using configuration file /etc/ckan/default/ckan.ini
2020-09-29 13:52:34,179 INFO  [ckan.config.environment] Loading static files from public
2020-09-29 13:52:34,202 INFO  [ckan.config.environment] Loading templates from /usr/lib/ckan/default/src/ckan/ckan/templates
2020-09-29 13:52:34,339 INFO  [ckan.config.environment] Loading templates from /usr/lib/ckan/default/src/ckan/ckan/templates
2020-09-29 13:52:34,389 CRITI [ckan.lib.uploader] Please specify a ckan.storage_path in your config
                         for your uploads
2020-09-29 13:52:34,483 INFO  [ckan.cli.server] Running server localhost on port 5000
2020-09-29 13:52:35,032 INFO  [ckan.cli] Using configuration file /etc/ckan/default/ckan.ini
2020-09-29 13:52:35,032 INFO  [ckan.config.environment] Loading static files from public
2020-09-29 13:52:35,056 INFO  [ckan.config.environment] Loading templates from /usr/lib/ckan/default/src/ckan/ckan/templates
2020-09-29 13:52:35,195 INFO  [ckan.config.environment] Loading templates from /usr/lib/ckan/default/src/ckan/ckan/templates
2020-09-29 13:52:35,246 CRITI [ckan.lib.uploader] Please specify a ckan.storage_path in your config
                         for your uploads
2020-09-29 13:52:35,340 INFO  [ckan.cli.server] Running server localhost on port 5000
``` 
but nothing served on port 5000 - a long delay waiting for response, and timeout.
`sudo netstat -peanut |grep 5000` (if I've installed net-tools) confirms I have
`tcp        0      0 127.0.0.1:5000          0.0.0.0:*               LISTEN      1001       61677      14256/python3`

The only thing that jumps out is the ckan.storage_path absence is marked as "CRITI" above - it would be surprising if that's the problem, since CKAN hasn't aborted, but I've set up the datastore anyway by following https://docs.ckan.org/en/2.9/maintaining/datastore.html and https://docs.ckan.org/en/2.9/maintaining/filestore.html - and the CRITI warning is now gone, but still no success on viewing a page on port 5000. (On that note, it may be good to mention the filestore vs datastore issue in Step 8 on the main installation page, rather than jumping for datastore)

Anyhow, at this point, I'm stuck and have nothing else to go on.

On my second try, I tried replacing solr-jetty with solr-tomcat in step 1, which then meant the instructions in step 5 were ok. I got exactly the same outcome though, with nothing served on 5000.

Thirdly, I read in a few places that solr-tomcat and solr-jetty both come with old versions of solr, so on the third try, I installed neither solr-jetty, nor solr-tomcat in Step 1, and replaced Step 5 with the instructions at https://github.com/ckan/ckan/wiki/Install-and-use-Solr-6.5-with-CKAN - the new Solr looks much more polished and I
could browse to it as expected on 8983 with the different solr_url, but again, nothing visible on port 5000 when I run the last step.

**Comment by weshinsley** [comment]
I should also have made clear, I was following the Python 3 instructions, and ignored all the 2.7 legacy ones.

**Comment by kowh-ai** [comment]
@weshinsley - I agree that the documentation is a bit confusing here (solr-jetty _vs_ solr-tomcat), however your issue of "nothing visible on port 5000" seems to be something else. At any point did you run the Section 7. commands (Create Database Tables)?

**Comment by weshinsley** [comment]
Yes, of course - I got the Initialising DB: SUCCESS in green.

**Comment by kowh-ai** [comment]
Yes if CKAN starts up without solr errors then solr should be OK

**Comment by weshinsley** [comment]
Can anyone confirm whether it's actually possible at present to install CKAN 2.9.0 from source into a new Ubuntu 20.04?

**Comment by kowh-ai** [comment]
@weshinsley - Yes Wes - I can confirm that installing CKAN 2.9.0 works as "install from source" and "install from package"

**Comment by weshinsley** [comment]
OK - thanks for confirming - I've done another reinstall, and I think I understand the issue now - I think the port 5000 server at the end of the instructions only listens on the loopback ethernet localhost/127.0.0.1, and not on the server's main IP address. 

So when installing specifically on a Ubuntu server edition (where you don't have a browser on the machine), you can't see the CKAN website (port 5000) by browsing to the machine's IP address, whereas you can do that for solr, which is listening on all ethernet interfaces.

Is that correct/does that make sense?

**Comment by Zharktas** [comment]
@weshinsley yes, the development server only listens on localhost by default. You can give parameter --host some_ip / 0.0.0.0 for it to listen other ones.

**Comment by weshinsley** [comment]
Many thanks @Zharktas  - I think that also could be added to the docs.

**Comment by weshinsley** [comment]
OK, so in summary, to address this issue I propose:

1) Change solr-jetty to solr-tomcat in step 1

2) In step 9, mention possibility of 
ckan -c /etc/ckan/default/ckan.ini run --host 0.0.0.0
should you need to browse from a different machine (eg, a ubuntu server install)

3) While at it, the ckan command-line documentation refers to ckan server, which I believe should be ckan run.

I will attempt to make a fork/simple PR for these next week. Thanks!

**Comment by jasonnykamp** [comment]
What does port 80 show for you? nginx should be passing everything from 80 to 5000.

**Comment by weshinsley** [comment]
nginx isn't involved at that point - I was running the development server in the 2.9.0 installation instructions, so only port 5000 (by default) is listened to. The instructions for nginx start afterwards, when you deploy the server. 

But the issue is fine - it's simply that (1) I didn't realise the development server was only listening on the loopback network interface, and (2) I was installing an ubuntu server which didn't have a browser. I was assuming that if you could theoretically browse on 127.0.0.1:5000, then you should also be able to browse to :5000 on the server's real ip address from another machine. Not so - unless you add the --host 0.0.0.0 on the development server, in which case you can.
